data-prepper-plugins/parquet-codecs/src/main/java/org/opensearch/dataprepper/plugins/codec/parquet/ParquetInputCodec.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.codec.parquet;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.parquet.ParquetReadOptions;
+import org.apache.parquet.column.page.PageReadStore;
+import org.apache.parquet.example.data.simple.SimpleGroup;
+import org.apache.parquet.example.data.simple.convert.GroupRecordConverter;
+import org.apache.parquet.hadoop.ParquetFileReader;
+import org.apache.parquet.hadoop.metadata.ParquetMetadata;
+import org.apache.parquet.hadoop.util.HadoopInputFile;
+import org.apache.parquet.io.ColumnIOFactory;
+import org.apache.parquet.io.MessageColumnIO;
+import org.apache.parquet.io.RecordReader;
+import org.apache.parquet.schema.MessageType;
+import org.apache.parquet.schema.Type;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.codec.InputCodec;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.log.JacksonLog;
+import org.opensearch.dataprepper.model.record.Record;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Consumer;
+
+/**
+ * An implementation of {@link InputCodec} which parses parquet records into fields.
+ */
+@DataPrepperPlugin(name = "parquet", pluginType = InputCodec.class)
+public class ParquetInputCodec implements InputCodec {
+
+    private static final String MESSAGE_FIELD_NAME = "message";
+    private static final String FILE_NAME = "parquet-data";
+    private static final String FILE_SUFFIX = ".parquet";
+
+    private static int fieldIndex;
+    private static final Logger LOG = LoggerFactory.getLogger(ParquetInputCodec.class);
+
+    @Override
+    public void parse(final InputStream inputStream, final Consumer<Record<Event>> eventConsumer) throws IOException {
+
+        Objects.requireNonNull(inputStream);
+        Objects.requireNonNull(eventConsumer);
+        parseParquetStream(inputStream, eventConsumer);
+
+    }
+
+    private void parseParquetStream(final InputStream inputStream, final Consumer<Record<Event>> eventConsumer) throws IOException {
+
+         final File tempFile = File.createTempFile(FILE_NAME, FILE_SUFFIX);
```

Comment:
We need to allow the pipeline author to specify a directory. In some cases, Data Prepper will not have access to the standard temp directory. So without that, Data Prepper may fail to run.

I made this comment on the other PR. Please see that comment for more details.

https://github.com/opensearch-project/data-prepper/pull/2309/files#r1148082689

Perhaps this can also come as a follow-on PR.